modules/thermal_hydraulics/src/materials/ADWallFrictionChengMaterial.C

```diff
@@ -0,0 +1,406 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ADWallFrictionChengMaterial.h"
+#include "WallFrictionModels.h"
+#include "SinglePhaseFluidProperties.h"
+#include "FlowModelSinglePhase.h"
+#include "Numerics.h"
+
+registerMooseObject("ThermalHydraulicsApp", ADWallFrictionChengMaterial);
+
+InputParameters
+ADWallFrictionChengMaterial::validParams()
+{
+  InputParameters params = Material::validParams();
+  params.addRequiredParam<MaterialPropertyName>("f_D", "Darcy friction factor material property");
+  params.addParam<MaterialPropertyName>(
+      "rho", FlowModelSinglePhase::DENSITY, "Density of the liquid");
```

Comment:
Is this friction factor only valid for liquid?

Reply:
It is a function of the P/D and Reynolds, so, I don't think the fluid type should matter if you're inside the correlation limits. I will replace for density of the fluid